src/dft/backends/mklcpu/commit.cpp

```diff
@@ -57,11 +57,11 @@ class commit_derived_impl final : public detail::commit_impl {
             throw oneapi::mkl::exception("dft/backends/mklcpu", "commit",
                                          "DftiCreateDescriptor failed");
         }
+    }
 
+    void commit(const detail::dft_values<prec, dom>& config_values) override {
         set_value(handle, config_values);
-
-        status = DftiCommitDescriptor(handle);
-        if (status != DFTI_NO_ERROR) {
+        if (auto status = DftiCommitDescriptor(handle); status != DFTI_NO_ERROR) {
```

Comment:
What is the point of this change? I found previous code easier to read.

Reply:
The status variable had to be declared anyway since this was the first use in this new function.
It was pretty silly though, because the status wasn't used so the return could have been directly compared.
I've used the status now, but I can split it into a definition and an if statement no problem.

Reply:
I have a small preference for moving the definition and assignment outside of the if block too.

Reply:
This can also be replaced by `DftiCommitDescriptor(handle) != DFTI_NO_ERROR` if we're not using the status elsewhere. I can see this is a common pattern in https://github.com/oneapi-src/oneMKL/pull/278 too.

Reply:
Updated in https://github.com/oneapi-src/oneMKL/pull/282/commits/ece0cc49ae480a551ec2aeb965369b40146e1e2c